### PR TITLE
Document ManualEventReader

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

- ManualEventReader is a bit confusing and complex

## Solution

- Document it.
- Also, rename `State` to `DoubleBufferState` for better clarity and CTRL+F usability.

### CODE REVIEW

File: crates/bevy_ecs/src/event.rs [hunk 1]
```diff
@@ -173,6 +176,15 @@ impl<'w, 's, T: Resource> EventWriter<'w, 's, T> {
...
 
+/// A simple event iterator which can be manually controlled.
+///
+/// In most cases, an [EventReader] will better meet your needs,
+/// as it both tracks which events a system has seen and fetches the relevant data.
+/// However, [EventReader] and [EventWriter] have conflicting accesses, and so cannot be used in the same system.
+///
+/// To get around this, you can store a [ManualEventReader] in a [Local] to help track events seen
+/// when working directly with `ResMut<Events<T>>`,
+/// allowing you to read and write events from within the same system.
```
Comment (mockersf) [inline crates/bevy_ecs/src/event.rs:187]: Should `ManualEventReader` be renamed given you can also "write" events with it?

Comment (mockersf) [inline crates/bevy_ecs/src/event.rs:187]: Reading this, I find it confusing. Why something named `ManualEventReader` would work better than using `EventReader`?. Maybe a code sample in doc could help to show how `ManualEventReader` works with `ResMut<Events<T>>` to provide the same functionality as a `EventReader` and a `EventWriter`?

**Comment by alice-i-cecile** [comment]
FYI, I'm a bit nervous about these docs: #2765 basically deprecates this usage pattern. If that is merged before this PR, the docs should be documented to reflect that (and I honestly wouldn't hate hiding `ManualEventReader` entirely).

**Comment by Weibye** [comment]
> FYI, I'm a bit nervous about these docs: #2765 basically deprecates this usage pattern. If that is merged before this PR, the docs should be documented to reflect that (and I honestly wouldn't hate hiding `ManualEventReader` entirely).

With #2765 merged, what is the status of this PR?

**Comment by alice-i-cecile** [comment]
Good call. I'm going to close this out.

**Pull request closed without merging**